src/util.ts

```diff
@@ -658,6 +658,42 @@ export async function handleFetch(request: string, options?: RequestInit) {
   return object;
 }
 
+/**
+ * Execute fetch and return object response, log if known error thrown, otherwise rethrow error.
+ *
+ * @param request - The request information.
+ * @param options - The fetch options.
+ * @param timeout - Timeout to fail request.
+ * @param errorCodesToCatch - array of error codes for errors we want to catch in a particular context
+ * @returns The fetch response JSON data or undefined (if error occurs).
+ */
+export async function fetchWithErrorHandling(
+  request: string,
+  options?: RequestInit,
+  timeout?: number | null,
```

Comment:
Would `number | undefined` make more sense here? I feel like `null` usually implies that `null` is intentional, whereas `undefined` usually implies that it's left out.

Reply:
fixed here: https://github.com/MetaMask/controllers/pull/805/commits/180b2802d9edef8ff9392b88ea3a3f76cc3373da